src/restic/node.go

```diff
@@ -191,7 +191,15 @@ func (node Node) createDirAt(path string) error {
 	return nil
 }
 
-func (node Node) createFileAt(path string, repo Repository) error {
+func (node Node) createFileAt(path string, repo Repository, idx *HardlinkIndex) error {
+	if node.Links > 1 && idx.Has(node.Inode, node.Device) {
+		err := fs.Link(idx.GetFilename(node.Inode, node.Device), path)
```

Comment:
ah, ok. and the current item not only has the reference (by name) back to the "hardlink master", but also the list of chunks, so it can be normally extracted?